broker/src/main/java/io/zeebe/broker/system/monitoring/BrokerHealthCheckService.java

```diff
@@ -83,4 +105,29 @@ private void initializePartitionInstallStatus() {
   public String getName() {
     return actorName;
   }
+
+  @Override
+  protected void onActorStarted() {
+    healthMonitor.startMonitoring();
+  }
+
+  private void registerComponent(final String componentName, final HealthMonitorable component) {
+    actor.run(() -> healthMonitor.registerComponent(componentName, component));
+  }
+
+  public void registerMonitoredPartition(final int partitionId, final HealthMonitorable partition) {
+    final String componentName = String.format(PARTITION_COMPONENT_NAME_FORMAT, partitionId);
+    registerComponent(componentName, partition);
+  }
+
+  public boolean isBrokerHealthy() {
+    return !actor.isClosed() && getBrokerHealth() == HealthStatus.HEALTHY;
+  }
+
+  private HealthStatus getBrokerHealth() {
+    if (isBrokerReady()) {
```

Comment:
Nit: it's maybe more read-able to express this as a guard clause, in which case you'd want to have the branch for the invariant and not the "common" case.

e.g.

```java
if (!isBrokerReady()) {
  return HealthStatus.UNHEALTHY;
}

return healthMonitor.getHealthStatus();
```